### PR TITLE
Cancelling reservations

### PR DESCRIPTION
**Pull request by jakubfijalkowski**
A simple command for cancelling the reservations by user

### CODE REVIEW

File: backend/src/Examples/ExampleApp.Examples.Domain/Booking/Reservation.cs [hunk 1]
```diff
@@ -56,6 +56,19 @@ public void Reject()
 
         Status = ReservationStatus.Rejected;
     }
+
+    public bool CanCancel => Status == ReservationStatus.Confirmed;
+
+    public void Cancel()
+    {
+        if (Status != ReservationStatus.Confirmed)
```
Comment (Saancreed) [inline backend/src/Examples/ExampleApp.Examples.Domain/Booking/Reservation.cs:64]:
```suggestion
        if (!CanCancel)
```

**Review by Saancreed — APPROVED** [review]
Out of curiosity, why did you choose to have attempts to release timeslots by other reservations silently do nothing?

File: backend/src/Examples/ExampleApp.Examples.Domain/Booking/Reservation.cs [hunk 2]
```diff
@@ -64,4 +77,5 @@ public enum ReservationStatus
     Confirmed = 1,
     Rejected = 2,
     Paid = 3,
+    Cancelled = 4, // New state
```
[the remark this answers is not in the document]
Comment (jakubfijalkowski) [reply backend/src/Examples/ExampleApp.Examples.Domain/Booking/Reservation.cs:80]: Hehe, Copilot :) Will remove ;)

File: backend/tests/Examples/ExampleApp.Examples.Tests/Handlers/Booking/FakeReservationsRepository.cs [hunk 3]
```diff
@@ -0,0 +1,5 @@
+using ExampleApp.Examples.Domain.Booking;
+
+namespace ExampleApp.Examples.Tests.Handlers.Booking;
+
+internal class FakeReservationsRepository : FakeRepositoryBase<Reservation, ReservationId> { }
```
[the remark this answers is not in the document]
Comment (jakubfijalkowski) [reply backend/tests/Examples/ExampleApp.Examples.Tests/Handlers/Booking/FakeReservationsRepository.cs:5]: Although I somewhat agree with you here, most of the code (apart from some `record`s) uses the `{ }` so I would leave that (and change everything at once if we decide to do so).

**Comment by jakubfijalkowski** [comment]
> Out of curiosity, why did you choose to have attempts to release timeslots by other reservations silently do nothing?

My thinking here was:
1) Handler runs, and commits, but message ACK fails,
2) The message is retried,
3) We don't want to fail, because that is a perfectly viable situation.

Now that I think about it, it can't happen, because we have a transactional outbox (and inbox) in place and it will prevent it first...

**Comment by jakubfijalkowski** [comment]
I'm leaving the "ignore" approach, but feel free @Saancreed to contest that (I doubt myself now) and I will change it.

**Pull request merged**